From 92facaf44aed64e3f33a5c919695c7dc59401c07 Mon Sep 17 00:00:00 2001
From: Chen Tianmi <tianmi.chen@intel.com>
Date: Tue, 30 Apr 2013 14:17:10 +0800
Subject: [PATCH 44/56] libstagefright: Ignore stsd box flag check for
 MPEG4Extractor

BZ: 50109

Ignore stsd box flag check to have more MP4 files played.

Change-Id: Ib400345a837f34d3a497c380511b134700de3055
Orig-Change-Id: Icbe9af5c8b1b9f046b0b8ddda9b48341261f8fae
Category: aosp improvement
Domain: Video.Media-container
Origin: internal
Upstream-Candidate: yes
Signed-off-by: Tianmi Chen <tianmi.chen@intel.com>
Signed-off-by: Dan Liang <dan.liang@intel.com>
Signed-off-by: Yuanjun Huang <yuanjun.huang@intel.com>
---
 media/libstagefright/MPEG4Extractor.cpp | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/media/libstagefright/MPEG4Extractor.cpp b/media/libstagefright/MPEG4Extractor.cpp
index b54de5c..bad9cf9 100644
--- a/media/libstagefright/MPEG4Extractor.cpp
+++ b/media/libstagefright/MPEG4Extractor.cpp
@@ -1425,11 +1425,6 @@ status_t MPEG4Extractor::parseChunk(off64_t *offset, int depth) {
                 return ERROR_IO;
             }
 
-            if (U32_AT(buffer) != 0) {
-                // Should be version 0, flags 0.
-                return ERROR_MALFORMED;
-            }
-
             uint32_t entry_count = U32_AT(&buffer[4]);
 
             if (entry_count > 1) {
-- 
2.7.4

